### PR TITLE
Fixes #10333: Document how to build a and agent

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/10333

### CODE REVIEW

File: 4_advanced_usage/18_build_agent.txt [hunk 1]
```diff
@@ -0,0 +1,72 @@
+==== Building the Rudder Agent
+
+===== Get source
+
+Make sure you have network access and the git command.
+
+Got to your build directory and checkout rudder-packages
```
Comment (ncharles) [inline 4_advanced_usage/18_build_agent.txt]: Typo: Go

File: 4_advanced_usage/18_build_agent.txt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+Choose the branch to build
+```
+# or any branch >= 4.1
+git checkout branches/rudder/4.1
+cd rudder-packages/rudder-agent
+```
+
+Now choose one of the 3 next chapter, depending on your case : debian package, redhat package or other.
```
Comment (ncharles) [inline 4_advanced_usage/18_build_agent.txt]: next 3 chapters
on your case:
(no space before colon)

File: 4_advanced_usage/18_build_agent.txt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+Got to your build directory and checkout rudder-packages
+
+```
+cd /usr/src
+git clone https://github.com/Normation/rudder-packages.git
+```
+
+Choose the branch to build
+```
+# or any branch >= 4.1
```
Comment (ncharles) [inline 4_advanced_usage/18_build_agent.txt]: this comment is misleading - it should be "For branch 4.1"

File: 4_advanced_usage/18_build_agent.txt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+Now choose one of the 3 next chapter, depending on your case : debian package, redhat package or other.
+
+===== Build a debian package
+
+Set the version to build:
+
+* Update the debian/changelog file to make the first entry match the version you want to build.
+* Edit the SOURCES/Makefile file and put a value into RUDDER_VERSION_TO_PACKAGE: see http://www.rudder-project.org/archives/ for a complete list of available versions.
```
Comment (ncharles) [inline 4_advanced_usage/18_build_agent.txt]: and set the value of RUDDER_VERSION_TO_PACKAGE

File: 4_advanced_usage/18_build_agent.txt [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+The package will be stored in the parent directory.
+
+===== Build a redhat package
+
+Run the redhat package builder
+
+```
+# TODO this is not enough
```
Comment (ncharles) [inline 4_advanced_usage/18_build_agent.txt]: Indeed, this is not enough :)

File: 4_advanced_usage/18_build_agent.txt [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+rpm -ba SPECS/*.spec
+```
+
+===== Build an install localy
+
+Before building the agent, you must decide on some environment variables :
```
Comment (ncharles) [inline 4_advanced_usage/18_build_agent.txt]: no space before :

File: 4_advanced_usage/18_build_agent.txt [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+* USE_SYSTEM_PCRE: (default true), use system pcre (depends on libpcre3-dev) or build it with the agent.
+
+* USE_SYSTEM_PERL: (default false), use system perl (depends on perl) or build it with the agent.
+
+* USE_SYSTEM_FUSION: (default false), use system fusion (depends on fusioninventory-agent), or build it with the agent. We advise you to use the Rudder version since it contains some patches.
+
+```
+# example
+env="RUDDER_VERSION_TO_PACKAGE=4.1.0~beta3 DESTDIR=/ USE_SYSTEM_PERL=true"
```
Comment (ncharles) [inline 4_advanced_usage/18_build_agent.txt]: could you document with a non beta version ?

**Comment by peckpeck** [comment]
Commit modified

File: 4_advanced_usage/18_build_agent.txt [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+```
+cd /usr/src
+git clone https://github.com/Normation/rudder-packages.git
+cd rudder-packages
+```
+
+Choose the branch to build
+```
+# For branch 4.1 (branches before 4.1 are not rupported)
```
Comment (amousset) [inline 4_advanced_usage/18_build_agent.txt]: supported

File: 4_advanced_usage/18_build_agent.txt [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+git checkout branches/rudder/4.1
+cd rudder-agent
+```
+
+Now choose one of the 3 next chapter, depending on your case: debian package, redhat package or other.
```
Comment (amousset) [inline 4_advanced_usage/18_build_agent.txt]: More precisely dpkg or rpm

File: 4_advanced_usage/18_build_agent.txt [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+===== Build a redhat package
+
+Run the redhat package builder
+
+Set the version to build:
+* Edit the SOURCES/Makefile file and set the value of RUDDER_VERSION_TO_PACKAGE: see http://www.rudder-project.org/archives/ for a complete list of available versions.
+
+```
+# make sure you are un in rudder-agent, then
```
Comment (amousset) [inline 4_advanced_usage/18_build_agent.txt]: you are in

**Comment by peckpeck** [comment]
Commit modified

File: 4_advanced_usage/18_build_agent.txt [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+Run the debian package builder
+
+```
+dpkg-buildpackage
+```
+
+The package will be stored in the parent directory.
+
+===== Build a redhat package
```
Comment (amousset) [inline 4_advanced_usage/18_build_agent.txt]: rpm

File: 4_advanced_usage/18_build_agent.txt [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+Choose the branch to build
+```
+# For branch 4.1 (branches before 4.1 are not supported)
+git checkout branches/rudder/4.1
+cd rudder-agent
+```
+
+Now choose one of the 3 next chapter, depending on your case: dpkg (debian like package), rpm (redhat like package) or other.
+
+===== Build a debian package
```
Comment (amousset) [inline 4_advanced_usage/18_build_agent.txt]: dpkg

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**